src/pytorch_lightning/core/module.py

```diff
@@ -569,7 +575,9 @@ def log_grad_norm(self, grad_norm_dict):
         """
         self.log_dict(grad_norm_dict, on_step=True, on_epoch=True, prog_bar=False, logger=True)
 
-    def all_gather(self, data: Union[Tensor, Dict, List, Tuple], group: Optional[Any] = None, sync_grads: bool = False):
+    def all_gather(
+        self, data: Union[Tensor, Dict, List, Tuple], group: Optional[Any] = None, sync_grads: bool = False
+    ) -> Tensor:
```

Comment:
I don't think it returns `Tensor`. Should return the type of data.

Reply:
by `type of data`, do you mean `Union[Tensor, Dict, List, Tuple]`

Reply:
yes

Reply:
updated in https://github.com/Lightning-AI/lightning/pull/13603/commits/b9ed56e9b7898b61b6491292412a4ec1dd0b8df7